libs/vkd3d/vkd3d_private.h

```diff
@@ -1142,6 +1143,20 @@ struct d3d12_bind_point_layout
     VkShaderStageFlags vk_push_stages;
 };
 
+#define VKD3D_MAX_HOISTED_DESCRIPTORS 4
```

Comment:
Do we know how many CBVs apps typically use in a pipeline? 4 seems very low; at least in D3D11 it's not uncommon to have 8-12 CBVs used in the vertex shader and then another bunch in the fragment shader.

Even if we end up triggering the descriptor set fallback (rather than push descriptors) when exceeding driver limits, this should still provide an uplift on NV.

Reply:
I guess we can bump this. I am a bit worried about push descriptor spam if root signature has a lot of descriptors and shaders only use a few anyways. 16 sounds fair?